From fe2fa43cdc5b6a9ede5151f362d29082e0db3fe5 Mon Sep 17 00:00:00 2001
From: Christian Marangi <ansuelsmth@gmail.com>
Date: Wed, 15 Nov 2023 00:57:41 +0100
Subject: [PATCH 3/6] hsl_phy: add support for detection PSGMII PHY mode

Add support for detection of PSGMII PHY mode to correctly detect qca807x
PHY upstream driver.

Signed-off-by: Christian Marangi <ansuelsmth@gmail.com>
---
 src/hsl/phy/hsl_phy.c | 3 +++
 1 file changed, 3 insertions(+)

diff --git a/src/hsl/phy/hsl_phy.c b/src/hsl/phy/hsl_phy.c
index 816f936b..399a85b9 100644
--- a/src/hsl/phy/hsl_phy.c
+++ b/src/hsl/phy/hsl_phy.c
@@ -1335,6 +1335,9 @@ hsl_port_phydev_interface_mode_status_get(a_uint32_t dev_id, a_uint32_t port_id,
 		case PHY_INTERFACE_MODE_10GKR:
 			*interface_mode_status = PORT_10GBASE_R;
 			break;
+		case PHY_INTERFACE_MODE_PSGMII:
+			*interface_mode_status = PHY_PSGMII_BASET;
+			break;
 		case PHY_INTERFACE_MODE_QSGMII:
 			*interface_mode_status = PORT_QSGMII;
 			break;
-- 
2.43.2

